source/crc.c

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include "checksums.h"
+
+#include "aws/checksums/crc.h"
+#include "aws/common/byte_buf.h"
+
+PyObject *aws_py_checksums_crc32(PyObject *self, PyObject *args) {
+    (void)self;
+
+    Py_buffer input;
```

Comment:
I made a small comment about this in the last review, but it got vanished so I'll make it again:
Py_buffer needs to be cleaned up if `PyArg_ParseTuple()` succeeds.

Lots of things in C that get created successfully, need to be manually "cleaned up" or "released" when you're done with them. Things like `int` don't need to be cleaned up because they're just some bytes on the stack. Things like `PyObject *` that we get from `PyArg_ParseTuple()` don't need to be cleaned up because they're just passed into this function by pointer (it's a "borrow" in Rust terms). But `Py_buffer` is different. Notice that it's by-value and not by-pointer. If `PyArg_ParseTuple()` succeeds, then it has "initialized" `Py_buffer` in a way that it needs to be cleaned up when we're done with it. How does a beginner learn this subtle detail? They read a [lot](https://docs.python.org/3/extending/extending.html) [of](https://docs.python.org/3/c-api/arg.html#strings-and-buffers) [docs](https://docs.python.org/3/c-api/buffer.html) 😬. It needs to be cleaned up whether `aws_py_checksums_crc32()` succeeds or fails. And this is why `goto` is popular in C. `goto` is not popular in any other language because in any other language you can make cleanup happen automatically as part of a type's destructor. Yeah ... there's a reason people don't write C anymore.

You're going to want the function to look something like this (incoming pseudocode!)
```
    Py_buffer input; /* Py_buffers must be released after successful PyArg_ParseTuple() calls */
    if (!PyArg_ParseTuple(...))
        return NULL

    /* From hereon, we need to clean up before returning */
    PyObject *py_result = NULL;

    ...
    If (something went wrong like the contiguous check, or the int checks)
        goto done;
    ...

   
    py_result = PyLong_FromUnsignedLong(crc_res);
done:
    PyBuffer_Release(&payload_buf);
    return py_result;
```